nlp/gender_debiasing_cda/gender_debiasing_cda.ipynb

```diff
@@ -0,0 +1,1032 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 15,
+      "metadata": {
+        "id": "IgyGIFcmryAa"
+      },
+      "outputs": [],
+      "source": [
+        "gender_pairs = [\n",
+        "      (\"girl\", \"boy\"),\n",
+        "      (\"she\", \"he\"),\n",
+        "      (\"woman\", \"man\")]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 16,
+      "metadata": {
+        "id": "LvDKyNVy5jvr"
+      },
+      "outputs": [],
+      "source": [
+        "gender_vectors = []\n",
+        "for (female_word, male_word) in gender_pairs:\n",
+        "    gender_vectors.append(vec_wikibio[female_word] - vec_wikibio[male_word])\n",
+        "    gender_vectors.append(vec_wikibio[male_word] - vec_wikibio[female_word])\n",
+        "\n",
+        "pca = PCA(n_components=1)\n",
+        "pca.fit(np.array(gender_vectors))\n",
+        "\n",
+        "female_vector = np.mean(\n",
+        "  pca.transform(np.array([vec_wikibio[pair[0]] for pair in gender_pairs]))\n",
+        ")\n",
+        "male_vector = np.mean(\n",
+        "  pca.transform(np.array([vec_wikibio[pair[1]] for pair in gender_pairs]))\n",
+        ")\n",
+        "mean_projection = (male_vector + female_vector) / 2"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 17,
+      "metadata": {
+        "id": "4qbI_g7_rvfx"
+      },
+      "outputs": [],
+      "source": [
+        "gender_vectors_unbiased = []\n",
+        "for (female_word, male_word) in gender_pairs:\n",
+        "    gender_vectors_unbiased.append(vec_unbiased_wikibio[female_word] - vec_unbiased_wikibio[male_word])\n",
+        "    gender_vectors_unbiased.append(vec_unbiased_wikibio[male_word] - vec_unbiased_wikibio[female_word])\n",
+        "\n",
+        "pca_unbiased = PCA(n_components=1)\n",
+        "pca_unbiased.fit(np.array(gender_vectors_unbiased))\n",
+        "\n",
+        "female_vector_unbiased = np.mean(\n",
+        "  pca_unbiased.transform(np.array([vec_unbiased_wikibio[pair[0]] for pair in gender_pairs])) \n",
+        ")\n",
+        "male_vector_unbiased = np.mean(\n",
+        "  pca_unbiased.transform(np.array([vec_unbiased_wikibio[pair[1]] for pair in gender_pairs]))\n",
+        ")\n",
+        "mean_projection_unbiased = (male_vector_unbiased + female_vector_unbiased) / 2"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 18,
+      "metadata": {
+        "id": "O2MBLLoDr_WJ"
+      },
+      "outputs": [],
+      "source": [
+        "jobs = [\"singer\", \"teacher\", \"doctor\", \"pilot\", \"developer\",  \"lawyer\",  \"coach\", \"engineer\", 'scientist']"
```

Comment:
Some elements have two whitespaces between comma and element (lawyer, coach)

Reply:
Does it make sense to include a couple more _jobs_ in this list, such that the correlation in the raw dataset is not always negative (only singer is positive)?

Reply:
Apparently it wasn't that easy to find one, this might be a shortcoming in the data indeed. Added "fashion" as an example now